compiler/noirc_frontend/src/tests.rs

```diff
@@ -1726,9 +1723,71 @@ fn numeric_generic_as_param_type() {
     ));
     // Error from the return type
     assert!(matches!(
-        errors[2].0,
+        errors[1].0,
```

Comment:
```suggestion
        errors[2].0,
```
This should still be 2 no?